empyrical/stats.py

```diff
@@ -20,11 +20,102 @@
 from scipy import stats
 from six import iteritems
 
-from .utils import nanmean, nanstd, nanmin, up, down, roll
+from .utils import nanmean, nanstd, nanmin, up, down, roll, rolling_window
 from .periods import ANNUALIZATION_FACTORS, APPROX_BDAYS_PER_YEAR
 from .periods import DAILY, WEEKLY, MONTHLY, YEARLY
 
 
+def _create_unary_vectorized_roll_function(function):
+    def unary_vectorized_roll(arr, window, out=None, **kwargs):
+        """
+        Computes the {human_readable} measure over a rolling window.
+
+        Parameters
+        ----------
+        arr : array-like
+            The array to compute the rolling {human_readable} over.
+        window : int, required
+            Size of the rolling window in terms of the periodicity of the data.
+        out : array-like, optional
+            The array to store the store the output.
+        **kwargs
+            Forwarded to :func:`~empyrical.{name}`.
+
+        Returns
+        -------
+        rolling_{name} : array-like
+            The rolling {human_readable}.
+        """
+        allocated_output = out is None
+
+        if len(arr):
+            out = function(rolling_window(arr, window).T, out=out, **kwargs)
+        else:
+            out = np.empty(0, dtype='float64')
+
+        if allocated_output and isinstance(arr, pd.Series):
+            out = pd.Series(out)
+
+        return out
+
+    unary_vectorized_roll.__doc__ = unary_vectorized_roll.__doc__.format(
+        name=function.__name__,
+        human_readable=function.__name__.replace('_', ' '),
+    )
+
+    return unary_vectorized_roll
+
+
+def _create_binary_vectorized_roll_function(function):
+    def binary_vectorized_roll(lhs, rhs, window, out=None, **kwargs):
+        """
+        Computes the {human_readable} measure over a rolling window.
+
+        Parameters
+        ----------
+        lhs : array-like
+            The first array to pass to the rolling {human_readable}.
+        rhs : array-like
+            The second array to pass to the rolling {human_readable}.
+        window : int, required
+            Size of the rolling window in terms of the periodicity of the data.
+        **kwargs
+            Forwarded to :func:`~empyrical.{name}`.
+
+        Returns
+        -------
+        rolling_{name} : array-like
+            The rolling {human_readable}.
+        """
+        allocated_output = out is None
+
+        if window >= 1 and len(lhs) and len(rhs):
+            out = function(
+                rolling_window(lhs, window).T,
```

Comment:
I'm a bit confused why the rolling window is being transposed.

Reply:
Because the functions allow dataframes, axis 0 is the time dimension and axis 1 are the unique time series. Rolling window adds the new dimension to the front, so we need to transpose.